clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -890,6 +894,11 @@ public void subscribe(Collection<String> topics, ConsumerRebalanceListener liste
                     if (topic == null || topic.trim().isEmpty())
                         throw new IllegalArgumentException("Topic collection to subscribe to cannot contain null or empty topic");
                 }
+
+                if (assignors.isEmpty())
+                    throw new IllegalStateException("Must configure at least one partition assigner class name to" +
+                        " partition.assignment.strategy config");
```

Comment:
nit: Can we reference the property in `ConsumerConfig` instead? Also, not sure it's worth it, but we could factor out a function, say `throwIfNoAssignorsConfigured` to avoid duplicating the comment.

Reply:
sounds good..Updated PR